src/panels/config/integrations/integration-panels/zha/dialog-zha-migrate-channel.ts

```diff
@@ -0,0 +1,149 @@
+import { html, LitElement, TemplateResult, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import { stopPropagation } from "../../../../../common/dom/stop_propagation";
+import { HassDialog } from "../../../../../dialogs/make-dialog-manager";
+import { changeZHANetworkChannel } from "../../../../../data/zha";
+import { showAlertDialog } from "../../../../../dialogs/generic/show-dialog-box";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import { HomeAssistant } from "../../../../../types";
+import "../../../../../components/buttons/ha-progress-button";
+import "../../../../../components/ha-button";
+import "../../../../../components/ha-select";
+import "@material/mwc-list/mwc-list-item";
+import { ZHAMigrateChannelDialogParams } from "./show-dialog-zha-migrate-channel";
+
+const VALID_CHANNELS = [
+  "auto",
+  11,
+  12,
+  13,
+  14,
+  15,
+  16,
+  17,
+  18,
+  19,
+  20,
+  21,
+  22,
+  23,
+  24,
+  25,
+];
+
+@customElement("dialog-zha-migrate-channel")
+class DialogZHAMigrateChannel extends LitElement implements HassDialog {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _migrationInProgress = false;
+
+  @state() private _params?: ZHAMigrateChannelDialogParams;
+
+  private _newChannel: "auto" | number = "auto";
+
+  public async showDialog(
+    params: ZHAMigrateChannelDialogParams
+  ): Promise<void> {
+    this._params = params;
+  }
+
+  public closeDialog(): void {
+    this._params = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render(): TemplateResult | typeof nothing {
+    if (!this._params) {
+      return nothing;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        scrimClickAction
+        escapeKeyAction
+        @closed=${this.closeDialog}
+        .heading=${createCloseHeading(
+          this.hass,
+          this.hass.localize("ui.panel.config.zha.change_channel_dialog.title")
+        )}
+      >
+        <p>
+          ${this.hass.localize(
+            "ui.panel.config.zha.change_channel_dialog.migration_warning"
+          )}
+        </p>
+
+        <p>
+          <ha-select
+            .label=${this.hass.localize(
+              "ui.panel.config.zha.change_channel_dialog.new_channel"
+            )}
+            fixedMenuPosition
+            naturalMenuWidth
+            @selected=${this._newChannelChosen}
+            @closed=${stopPropagation}
+            value=${String(this._params.currentChannel)}
```

Comment:
```suggestion
            .value=${String(this._params.currentChannel)}
```

Reply:
The value should be the selected value, so should be `this._newChannel || this._params.currentChannel`